Firmware/MotorControl/low_level.c

```diff
@@ -1379,3 +1379,10 @@ void motor_thread(void const * argument) {
     }
     motor->thread_ready = false;
 }
+
+/** Function that sets the current encoder count to a desired 32-bit value. */
+void setEncoderCount(Motor_t* motor, uint32_t count){
+    motor->encoder.encoder_state = count;
+    motor->motor_timer->Instance->CNT = count;
```

Comment:
There is a race condition hazard here: If this function is running on a low priority thread (say comms thread), it could in an unfortunate timing event get preempted right between these two lines, and have the motor thread's update_rotor run in-between. That would not be good.
We can make a short critical-section. We can check if the current OS api has an abstracted critical section. Otherwise, critical sections are usually implemented disable all interrupts (global interrupt mask), some critical code, enable them again.